pkg/datagatherer/k8s/client_test.go

```diff
@@ -0,0 +1,77 @@
+package k8s
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	clientcmdlatest "k8s.io/client-go/tools/clientcmd/api/latest"
+)
+
+// These tests do not currently validate the created dynamic client uses the
+// KUBECONFIG file that we create, however it _does_ exercise enough of the
+// code path to show that the function is correctly selecting which file to
+// load and returning it.
+
+func TestNewDynamicClient_ExplicitKubeconfig(t *testing.T) {
+	kc := createValidTestConfig()
+	path := writeConfigToFile(t, kc)
+	_, err := NewDynamicClient(path)
+	if err != nil {
+		t.Error("failed to create client: ", err)
+	}
+}
+
+func TestNewDynamicClient_InferredKubeconfig(t *testing.T) {
+	kc := createValidTestConfig()
+	path := writeConfigToFile(t, kc)
+	cleanupFn := temporarilySetEnv("KUBECONFIG", path)
+	defer cleanupFn()
+	_, err := NewDynamicClient("")
+	if err != nil {
+		t.Error("failed to create client: ", err)
+	}
+}
+
+func writeConfigToFile(t *testing.T, cfg clientcmdapi.Config) string {
+	f, err := ioutil.TempFile("", "testcase-*")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	if err := clientcmdlatest.Codec.Encode(&cfg, f); err != nil {
+		t.Fatal(err)
+	}
+	return f.Name()
+}
+
+func createValidTestConfig() clientcmdapi.Config {
+	const (
+		server = "https://anything.com:8080"
```

Comment:
nit but can we use example.com?

Reply:
sure, this was taken straight from client-go so didn't pay much attention :)